tesseract/tesseract_planning/tesseract_motion_planners/include/tesseract_motion_planners/descartes/impl/descartes_collision_edge_evaluator.hpp

```diff
@@ -0,0 +1,169 @@
+ * @par
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ * http://www.apache.org/licenses/LICENSE-2.0
+ * @par
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+#ifndef TESSERACT_MOTION_PLANNERS_IMPL_DESCARTES_COLLISION_EDGE_EVALUATOR_HPP
+#define TESSERACT_MOTION_PLANNERS_IMPL_DESCARTES_COLLISION_EDGE_EVALUATOR_HPP
+
+#include <tesseract/tesseract.h>
+TESSERACT_COMMON_IGNORE_WARNINGS_PUSH
+#include <numeric>
+TESSERACT_COMMON_IGNORE_WARNINGS_POP
+
+#include <tesseract_motion_planners/descartes/descartes_collision_edge_evaluator.h>
+#include <tesseract_environment/core/utils.h>
+
+namespace tesseract_motion_planners
+{
+template <typename FloatType>
+DescartesCollisionEdgeEvaluator<FloatType>::DescartesCollisionEdgeEvaluator(
+    const tesseract_environment::Environment::ConstPtr& collision_env,
+    std::vector<std::string> active_links,
+    std::vector<std::string> joint_names,
+    double collision_safety_margin,
+    double longest_valid_segment_length,
+    bool allow_collision,
+    bool debug)
+  : state_solver_(collision_env->getStateSolver())
+  , acm_(*(collision_env->getAllowedCollisionMatrix()))
+  , active_link_names_(std::move(active_links))
+  , joint_names_(std::move(joint_names))
+  , discrete_contact_manager_(collision_env->getDiscreteContactManager())
+  , continuous_contact_manager_(collision_env->getContinuousContactManager())
+  , collision_safety_margin_(collision_safety_margin)
+  , longest_valid_segment_length_(longest_valid_segment_length)
+  , allow_collision_(allow_collision)
+  , debug_(debug)
+  , dof_(joint_names_.size())
+{
+  discrete_contact_manager_->setActiveCollisionObjects(active_link_names_);
+  discrete_contact_manager_->setContactDistanceThreshold(collision_safety_margin_);
+  discrete_contact_manager_->setIsContactAllowedFn(
+      std::bind(&tesseract_motion_planners::DescartesCollisionEdgeEvaluator<FloatType>::isContactAllowed,
+                this,
+                std::placeholders::_1,
+                std::placeholders::_2));
+
+  continuous_contact_manager_->setActiveCollisionObjects(active_link_names_);
+  continuous_contact_manager_->setContactDistanceThreshold(collision_safety_margin_);
+  continuous_contact_manager_->setIsContactAllowedFn(
+      std::bind(&tesseract_motion_planners::DescartesCollisionEdgeEvaluator<FloatType>::isContactAllowed,
+                this,
+                std::placeholders::_1,
+                std::placeholders::_2));
+}
+
+template <typename FloatType>
+bool DescartesCollisionEdgeEvaluator<FloatType>::evaluate(
```

Comment:
Not saying we shouldn't merge it, since I know there is an active need for this. But will any of our unit tests check that this is doing what it should?

Reply:
I added it to the descartes unit tests.